### ISSUE TITLE
List USDT (Tether)

### ISSUE THREAD

**Issue by flix1** [bisq-network/proposals#170]
> _This is a Bisq Network proposal. Please familiarize yourself with the [submission and review process](https://docs.bisq.network/proposals.html)._

### Background
USDT ([Tether](https://en.bitcoinwiki.org/wiki/Tether)) is a stablecoin based on [omnilayer](https://en.bitcoinwiki.org/wiki/Omni_Layer) Bitcoin coloured coin protocol and pegged 1:1 to the USD. The peg is maintained by a **centralized** institution which holds currency reserves and discloses these amounts here https://wallet.tether.to/transparency.

However trading and wallets are decentralized and based on an open protocol.

### Summary
USDT is the highest volume crypto, surpassing even Bitcoin in daily volume. It is traded on hundreds of [exchanges](https://coinmarketcap.com/currencies/tether/markets/). 

### Verification
USDT can be verified with multiple blockchain explorers like https://omniexplorer.info/. As a Bitcoin coloured coin all transactions are public and transparent.

### Proposal

- Exempt USDT from the current Bisq policy of putting new asset listings on hold.
- Merge PR to add USDT into Bisq in the same format as BSQ with "altcoin" and "altcoin instant" as payment methods.

**Comment by MwithM** [comment]
Agree with adding Tether, as long as it's not called a cryptocurrency and alert about possible risks. It's as centralized and obscure as any state fiat currency, maybe even more.

New asset listing is on hold for a reason. Either we wait until this hold is lifted or we say clearly that this hold is permanent and only for assets we dislike. I prefer the first option.

**Comment by flix1** [comment]
Apparently USDT was already available on Bisq and was removed because it was never traded:
https://bisq.community/t/new-version-0-6-0-released/3446

**Comment by flix1** [comment]
Actually @ripcurlx confirms it was never added:
https://github.com/bisq-network/bisq/pull/2395

**Comment by ripcurlx** [comment]
The problem why I didn't add Tether back then (https://github.com/bisq-network/bisq/pull/2395) was that there are two implementations (ERC-20, Omni) and I just didn't want to spend the time to implement two variations and to think about how we should communicate that in the best possible way.

**Comment by flix1** [comment]
I think only the omni one is worth it. As a bitcoin based coloured coin it could in the future allow single transaction trustless exchanges just like we were considering for BSQ in #50

**Comment by wiz** [comment]
> The problem why I didn't add Tether back then (bisq-network/bisq#2395) was that there are two implementations (ERC-20, Omni) and I just didn't want to spend the time to implement two variations

Right, and now there are 3 implementations, the third being L-USDT on Liquid. Since that has CT it surely has the best privacy how about we amend this proposal to add the Liquid Tether USD token?

**Comment by flix1** [comment]
If we want to get this done soon (next release) and cause the minimum amount of confusion I suggest that we pick one implementation. There's no point to duplicating efforts before we know if there is any demand for it.

I would bet on the Omni USDT implementation, as I think it is the oldest and more stable... but I am open to being convinced otherwise.

**Comment by wiz** [comment]
@flix1 we started talking about this again today... maybe we can add USDT and just have checkboxes for what networks you want to trade the token on, I understand there are many including Liquid, Omni, ERC20, TRC20, etc. so we could start with those 4 checkboxes and offers that aren't matched would be greyed out, similar to how fiat payment methods can be greyed out for the same fiat currency?

**Comment by flix1** [comment]
@wiz we should consider the possibility that this will have zero demand. So I would keep it simple and start with only one option, leaving the door open to others if people actually ask for it.

**Comment by ripcurlx** [comment]
The problem is not so much about the different implementations rather that the price ticker is not working for it out-of-the-box. So only the last trade price from Bisq will be the market price. Fixed pricepoints would need to be set in BTC, which makes it problematic if you have to keep the offer open for some time until it is taken.

**Comment by ricardosaurio** [comment]
Disagree, Bisq do needs a stablecoin but USDT is sketchy at best. I would suggest either DAI wich has alredy been added to be able to be traded permanently. At least it is semi-decentralized.

**Comment by MwithM** [comment]
Closing as stalled (can be reopened)

**Comment by flix1** [comment]
I think this issue should be reopened and seriously discussed for implementation in the next release (Bisq v1.3.8)

**Comment by pingiun** [comment]
I am in favour of adding USDT, the fact that DAI is more decentralized does not take away that there seems to be a large audience for USDT---DAI can always be added on a later day.

I propose that both the ERC20 and Omni versions be added, where the user can select which versions they want to be able to trade for their USDT account. I'm imagining this could work the same way you can select currencies for SEPA accounts.

**Comment by ripcurlx** [comment]
> I am in favour of adding USDT, the fact that DAI is more decentralized does not take away that there seems to be a large audience for USDT---DAI can always be added on a later day.

DAI was already added. It just was not traded at all. Everyone up for starting a market just needs to pay the listing fee to activate it again.

Regarding Tether: As mentioned in https://github.com/bisq-network/proposals/issues/170#issuecomment-602571305 the problem was back then that it didn't work out of the box without changing more parts of the code to support a price ticker for this. Maybe this changed with the new price providers, but that needs to be tested. In the past all stablecoins we added so far saw hardly any trading volume. So I think it only makes sense to add it if someone is willing to bootstrap the market for it.

**Comment by ripcurlx** [comment]
Proposal transaction ID: 947b72d7b321f03595decbf0ca86615e3e8057907e07b21d60c258ea978fab15

**Comment by flix1** [comment]
I think this issue should be reopened as per discussion on keybase.

>  I think this is important for Bisq:
> -Bisq is an exchange and survives thanks to trading fees
> -90% of volume on exchanges is done by traders. Not "regular users" who just buy or sell occasionally.
> -Traders need to book paper profits and therefore find stablecoins very useful. That is why USDT does so much volume.

**Comment by Mycelial1** [comment]
I am chiming in here because I am a bisq user that is in favor of re-listing DAI (or USDT) on bisq. I think DAI would be better because USDT has three or more implementions on different chains which is more difficult for devs to implement and can be confusing. think traders who want to take profits into usd are doing so at the risk of their bank asking questions like Flix pointed out. 

I would be very interested in having DAI (or something else) re-listed, and I would help however I can with growth efforts to promote it. I think it's a worthy endeavor that could bring much more volume to bisq.

**Comment by ripcurlx** [comment]
AFAIK DAI is already implemented. It just needs someone to pay the listing fee and do some market making so it generates enough volume to stay.

**Comment by MwithM** [comment]
Yes, it was relisted a few months ago. You can relist it at DAO - Asset listing fee/proof of burn.
Let the word spread if you relist it, I think it has chances to stay with enough volume.

**Comment by flix1** [comment]
![image](https://user-images.githubusercontent.com/4233784/110787244-699ee800-826d-11eb-88e9-00a22eb31b70.png)

I can make no assurance that USDT would have a high volume on Bisq. Maybe traders who use it would simply not be interested in the speed, liquidity, fees, ux of Bisq. But the **cost** of trying it is very low. The potential **profit** for Bisq is very high. I see making the attempt as a no brainer. If it fails, it fails... but nobody has yet to give me a good reason not to try. Even less to take more than a year of discussio

**Comment by BtcContributor** [comment]
Any update on this? Adding USDT/USDC would be easy I think and could lead to a lot of users.

**Comment by pazza83** [comment]
Re-opening this for discussion. Seems to be renewed interest in adding stable coins with Tether being preferred.

**Comment by leo816** [comment]
Yes, also believe we need to re-visit this discussion and I think bisq will benefit from having Tether.

**Comment by ricardosaurio** [comment]
as always, the market says something that the technology and the work behind it seems totally contradictory. This is an exchange and it will benefit the best from USDT but the shadiness behind USDT makes me to at least suggest DAI be listed along with USDT. At least DAI goes a lot better with the spirit and narrative of Bitcoin and Bisq

**Comment by clearwater-trust** [comment]
Listing  "_tokens of invisible tether_" coins would eventually expose their fraud. As it would only take a few trades outside their "**stable**" price to break the levy of fakery behind these scams. Plus, it would provide more revenue. Assuming people actually trade their fake tokens on Bisq. Generally, I am for adding new listings. 

I thought Bisq was aiming for a pay to play scheme when adding new "assets"?

**Comment by flix1** [comment]
I'm not going to repeat all the arguments in favour listed above. I think it will be good for Bisq.

However, given the amount of time that this issue has been unresolved I would ask those opposed to please state their reasons clearly here so that we can have an open debate.

If no-one is opposed I would please ask for no more delays and for this to be added in the next release.

If it helps... I also offer to pay the listing fee.

**Comment by pazza83** [comment]
> DAI be listed along with USDT. At least DAI goes a lot better with the spirit and narrative of Bitcoin and Bisq

DAI is already listed. Albeit not used that often.

USDC and TrueUSD are also listed already.

Maybe the UI of the offer book drop down list of currencies can be improved to make this clear. Think they might all need to be manually added by users at present.

**Comment by MwithM** [comment]
> At least DAI goes a lot better with the spirit and narrative of Bitcoin and Bisq

Until you know about [Dollar On Chain](https://moneyonchain.com/).
Adding this pegcoin at Bisq doesn't makes much sense, though. It's better to change BTC to RBTC at Bisq and then use the dapps at RSK (sovreign, MOC...) to exchange the RBTC to DOC.

**Comment by pazza83** [comment]
Seems there is consensus for adding USDT.

USDT was stalled last time due to multiple blockchains supporting it and no agreed decision of which to add.

To keep things simple, and avoid funds being lost with user confusion, I propose adding the ERC-20 version of UST. I am proposing the ERC-20 version as it has the largest holding of USDT. About 50% of USDT is held on ERC-20. Ref: https://wallet.tether.to/transparency

If USDT becomes popular on Bisq we could look at adding more versions.

I propose naming the first implementation of USDT as `USDT (ERC-20)`.

I am happy to add it as part of my Payment Methods Maintainer role.

**Comment by pazza83** [comment]
Approved. Many thanks to @jmacxx for completing this ready for version 1.7.5

@flix1 hope you can get some liquidity added to this market :)

**Comment by flix1** [comment]
> @flix1 hope you can get some liquidity added to this market :)

I will do my best to kickstart it and attract other traders.

**Comment by Akira45-0** [comment]
> @flix1 we started talking about this again today... maybe we can add USDT and just have checkboxes for what networks you want to trade the token on, I understand there are many including Liquid, Omni, ERC20, TRC20, etc. so we could start with those 4 checkboxes and offers that aren't matched would be greyed out, similar to how fiat payment methods can be greyed out for the same fiat currency?
> 
> EDIT: https://wallet.tether.to/transparency - so it seems we need 5 checkboxes for USDT, and let the users decide which ones they want to support

I totally agree. I am all for implementing USDT as a payment method the way wiz described. There will be no demand when there's no trading opportunity created by the availability of the payment method.

As for ERC-20 the Ethereum fees are exorbitant in comparison with fees on Liquid and Tron, so I would like to ask if there's a possibilty of implementing USDT in a "generic" way with checkboxes as wiz proposed.

I am interested in trading for USDT on Liquid and Tron sidechains, but if had to make separate offers for USDT Liquid and USDT Tron, then this would really be pointless.

BTW it does not work in the way "no demand for USDT ERC-20/Omni means no demand for USDT Tron/Liquid", as a lot of people will avoid transaction when they have to pay a few dozen $ only for the mining fees.
